### PR TITLE
Feature/goals UI and validation fixes

### LINKED ISSUE #457: Fix validation and UI bugs in goals
- Fix low-contrast font colors on `MyGoals` in dark mode
- Remove the list item indicator (the small square from the goal list)
- Add validation to avoid empty goals being added to the database
- Fix typescript bugs and workarounds

### PR DESCRIPTION
**Pull request by pulkitvyas08**
Resolves #457

### CODE REVIEW

**Comment by pulkitvyas08** [comment]
Since there was no design for showing validation errors I have added a standard placeholder

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]
Let's keep our conventions and hang-ups followed 😁

**Comment by pulkitvyas08** [comment]
Ready for review

File: src/pages/MyGoalsPage/MyGoalsPage.tsx [hunk 1]
```diff
@@ -9,13 +8,17 @@ import { archiveGoal, getActiveGoals, removeGoal, updateGoal, isCollectionEmpty
 import { GoalItem } from "@src/models/GoalItem";
 
 import "./MyGoalsPage.scss";
+import { useRecoilValue } from "recoil";
+import { darkModeState } from "@src/store";
```
[the remark this answers is not in the document]
Comment (pulkitvyas08) [reply src/pages/MyGoalsPage/MyGoalsPage.tsx:12]: Shit, my bad

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**